src/descriptionBuilder.js

```diff
@@ -0,0 +1,105 @@
+const { getEmojiUrl, imageToBase64 } = require("./utils");
+
+const DescriptionBuilder = function(desc, remove_emojis = false) {
+  let x = 25; // start x position of description
+  let maxX = 80; // description will be stripped if x exceeds 90 (= about 55 letters without emojis)
+  const y = 70; // vertical positioning of description line
+  let parentType = null; // text, image
+
+  // stores description parts
+  const descriptionRenderParts = [];
+
+  function addText(text, force = false) {
+    if (parentType === 'image') {
+      x += 2; // extra spacing between text and image
+    }
+    if (x+text.length > maxX && !force) {
+      text = `${text.slice(0, maxX-x)}..`;
+    }
+    descriptionRenderParts.push('<text class="description" x="' + x + '" y="' + y + '">' + text + '</text>');
+    x += text.length;
+    parentType = 'text';
+  }
```

Comment:
I think instead of mutating the descriptionParts array in here and pushing directly to it, we should make the _addText & _addImage function just return the svg text strings, and instead of calling then _addtText lets call them _createTextNode _createImageNode

Reply:
I somehow need to get the correct width of the text so that I can calculate `x` correctly.
Seems the only way to do that is to add the description section via DOM and get it with `getComputedTextLength()`.
So these lines are pretty sure to change. :D